frappe/public/js/frappe/list/list_view.js

```diff
@@ -1333,28 +1339,41 @@ frappe.views.ListView = class ListView extends frappe.views.BaseList {
 				return;
 			}
 
-			const { doctype, name } = data;
-			if (doctype !== this.doctype) return;
-
-			// filters to get only the doc with this name
-			const call_args = this.get_call_args();
-			call_args.args.filters.push([this.doctype, "name", "=", name]);
-			call_args.args.start = 0;
-
-			frappe.call(call_args).then(({ message }) => {
-				if (!message) return;
-				const data = frappe.utils.dict(message.keys, message.values);
-				if (!(data && data.length)) {
-					// this doc was changed and should not be visible
-					// in the listview according to filters applied
-					// let's remove it manually
-					this.data = this.data.filter((d) => d.name !== name);
-					this.render_list();
-					return;
-				}
+			this.pending_document_refreshes.push(data);
+			frappe.utils.debounce(this.process_document_refreshes.bind(this), 1000)();
+		});
+	}
+
+	process_document_refreshes() {
+		if (!this.pending_document_refreshes.length) return;
+
+		const names = this.pending_document_refreshes.map((d) => d.name);
+		this.pending_document_refreshes = this.pending_document_refreshes.filter(
+			(d) => names.indexOf(d.name) === -1
+		);
```

Comment:
@gavindsouza do you remember what this code was supposed to do?

Currently it just empties `this.pending_document_refreshes` in an inefficient way 😁

Reply:
@barredterra yeah, this isn't doing anything speical. I thought I removed it :see_no_evil:

Reply:
I remember discussing this with some folks in the framework team back then: The idea was not to overwrite the `pending_document_refreshes` queue while it may be modified by another thread. So, remove only the names that the current thread has read. 

Now, I don't know how instance attributes work with threads or speak of the validity of the scenario, I have lost touch with Desk & JS, this is just what I can gather from memory.